File: lab6/c/sp_algorithms.c
#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>
#include <string.h>
#include <stdarg.h>
#include <stdbool.h>
#include <assert.h>

#include "graph.h"
#include "pq.h"
#include "sp_algorithms.h"

sssp_result_t *bfs(graph_t const *g, node_t src, node_t dst) {
  unsigned long long stat_edges_explored=0;

  size_t N = graph_get_num_nodes(g);
  node_t *pred = CALLOC(node_t,N);
  weight_t *dist = CALLOC(weight_t,N);
    
  for (size_t i=0;i<N;++i) {
    pred[i]=INVALID_NODE;
    dist[i]=weight_inf();
  }

  error("Not implemented");

  return sssp_result_new(N,src,dst,false,pred,dist,stat_edges_explored);
}

sssp_result_t *bellman_ford(graph_t const *g, node_t src) {
  error("Not implemented");
}

sssp_result_t *dijkstra(graph_t const *g, node_t src, node_t dst) {
  error("Not implemented");
}

sp_result_t *astar_search(graph_t const *g, node_t src, node_t dst, weight_t const* h) {
  error("Not implemented");
}

File: lab6/python/sp_algorithms.py
import graph
import pq
import general
from shortest_path import sssp_result_t, sp_result_t
import weight

def bfs(g, src, dst):
    stat_edges_explored = 0
    
    N = g.graph_get_num_nodes()
    pred = [graph.INVALID_NODE]*N
    dist = [weight.weight_inf()]*N
    
    general.error("Not implemented")
    
    return sssp_result_t(N, src, dst, False, pred, dist, stat_edges_explored)
    
def bellman_ford(g, src):
    general.error("Not implemented")
    
def dijkstra(g, src, dst):
    general.error("Not implemented")
    
def astar_search(g, src, dst, h):
    general.error("Not implemented")
